cyences_app_for_splunk/bin/sophos_instance_details.py

```diff
@@ -0,0 +1,217 @@
+        data = json.loads(serverContent)['entry']
+        client_id = ''
+        client_secret = ''
+        for i in data:
+            if i['name'] == 'countermeasure_sophos':
+                client_id = i['content']['client_id']
+                client_secret = cs_utils.CredentialManager(sessionKey).get_credential(client_id)
+                break
+        return client_id,client_secret
+
+    def get_barier_token(self,client_id,client_secret):
+        header = {"Content-Type":"application/x-www-form-urlencoded"}
+        data = {'client_id':client_id,'client_secret':client_secret,'grant_type':'client_credentials','scope':'token'}
+        response = requests.post(SOPHOS_AUTH_URL, headers=header,data = data)
+
+        response_json = response.json()
+        if(response_json.get('errorCode')=='success'):
+            return response_json.get('access_token')
+        else:
+            self.logger.error("Error while fetching bearier Token : {}".format(response_json))
+            exit()
+
+    def get_tenant_from_organization(self,barier_token,organization_id):
+
+        header = {"Authorization":"Bearer {}".format(barier_token),"X-Organization-ID":organization_id}
+        response = requests.get(SOPHOS_TENANT_FROM_ORGANIZATION_URL, headers=header)
+
+        if(response.status_code == 200):
+            response_json = response.json()
+            current_page = 1
+            for i in response_json.get("items"):
+                SOPHOS_TENANT_DICT[i.get('id')] = i.get('apiHost')
+            total_pages = response_json.get('pages').get('total')
+            while current_page <= total_pages:
+                current_page=current_page+1
+                header = {"Authorization":"Bearer {}".format(barier_token),"X-Organization-ID":organization_id}
+                response = requests.get(SOPHOS_TENANT_FROM_ORGANIZATION_PAGINATION_URL + str(current_page), headers=header)
+                response_json = response.json()
+                for i in response_json.get("items"):
+                    SOPHOS_TENANT_DICT[i.get('id')] = i.get('apiHost')
+
+        else:
+            self.logger.error("Error while fetching Tenant Details: {}".format(response.json()))
+            exit()
+
+
+    def get_tenant_list(self,barier_token):
+
+        header = {"Authorization":"Bearer {}".format(barier_token)}
+        response = requests.get(SOPHOS_WHOAMI_URL, headers=header)
+
+        if(response.status_code == 200):
+            response_json = response.json()
+            if(response_json.get("idType")=="tenant"):
+                SOPHOS_TENANT_DICT[response_json.get("id")] = response_json.get("apiHosts").get("dataRegion")
+            elif(response_json.get("idType")=="organization"):
+                self.get_tenant_from_organization(barier_token,response_json.get("id"))
+        else:
+            self.logger.error("Error while fetching Tenant Details: {}".format(response.json()))
+
```

Comment:
raise error so that it can get notified in the search UI. If this error is not expected and you don't want to continue the rest of the code, otherwise this is fine.